packages/apollo-gateway/src/index.ts

```diff
@@ -126,11 +147,56 @@ export class ApolloGateway implements GraphQLService {
     this.isReady = true;
   }
 
+  public onSchemaChange(value: SchemaChangeCallback): Unsubscriber {
+    if (!isManagedConfig(this.config)) {
+      return () => {};
+    }
+
+    this.onSchemaChangeListeners.add(value);
+    if (!this.pollingTimer) this.startPollingServices();
+
+    return () => {
+      this.onSchemaChangeListeners.delete(value);
+      if (this.onSchemaChangeListeners.size === 0 && this.pollingTimer) {
+        clearInterval(this.pollingTimer!);
+        this.pollingTimer = undefined;
+      }
+    };
+  }
+
+  private startPollingServices() {
+    if (this.pollingTimer) clearInterval(this.pollingTimer);
+
+    this.pollingTimer = setInterval(async () => {
+      try {
+        const [services, isNewSchema] = await this.loadServiceDefinitions(
+          this.config,
+        );
+        if (!isNewSchema) {
+          this.logger.debug('No changes to gateway config');
+          return;
+        }
+        if (this.queryPlanStore) this.queryPlanStore.flush();
+        this.logger.debug('Gateway config has changed, updating schema');
+        this.createSchema(services);
+        this.onSchemaChangeListeners.forEach(listener =>
+          listener(this.schema!),
+        );
+      } catch (e) {
+        this.logger.debug(
+          'Error checking for schema updates. Falling back to existing schema.',
```

Comment:
This error message doesn't seem like it's necessarily true.  Depending on when the error occurred, `createSchema` may have already been called.

I'm also not sure if an error which occurs in `onSchemaChangeListeners` should necessarily prevent a schema from updating (perhaps it should actually rollback, but also perhaps that's just the fault of the subscriber and not us?  We're _just_ passing a schema.).  It certainly seems worth logging about it though!